### DISCUSSION TITLE
Is there any way to download repseqio or gene library automatically?

### DISCUSSION THREAD

**Discussion by omegahh** [milaboratory/mixcr#1531] [Q&A] [answered]
The gene library in vdj.online is very convenient through, for reproducible research and further genomic analysis, I really want to have the VDJC library used in MiXCR as a certain file. As what I can see is "repseqio.v4.0" from `mixcr --version`. Is it possible?

**Comment by mizraelson** [comment, marked as answer]
Hi, all genes from the library are present on our dedicated website: [https://vdj.online/library](https://vdj.online/library)

You can export all genes of interest in tsv or FASTA format.

**Comment by omegahh** [comment]
Thank you for replying. I do use gene library (https://vdj.online/library) since it was published. But I still see different version number of V/D/J/C library between mixcr 4.5.0 and mixcr 4.6.0, I wander what are the differences?
![截屏2024-02-11 15 20 06](https://github.com/milaboratory/mixcr/assets/5026687/ef6b9cdd-71ff-4782-a4a5-e796189cf5ff)
![截屏2024-02-11 15 20 21](https://github.com/milaboratory/mixcr/assets/5026687/3b195596-0823-459e-9435-ae2f6526b16f)
Actually, I have two questions:
1. How can I determine the authentic genes in a specific version of repseqio if the gene library lacks version information?
2. I've developed some algorithms that rely on V/J gene sequences. To ensure reproducible research, should I simply use the current published V/D/J/C gene names and sequences from the gene library? However, I'm aware that the gene library undergoes continuous updates.

**Reply by mizraelson** [reply]
There are minor differences between version 4.0 and 3.0.1 of our software, which primarily do not impact the names or sequences of the genes. Some allelic numbers have been updated to better align with the International Union of Immunological Societies (IUIS) standards, and adjustments have been made to the coordinates for the `UTR5Begin` and `VEnd` of V genes. The most up-to-date gene library is always available at [https://vdj.online/library](https://vdj.online/library). Additionally, you have the option to export a column containing the germline gene sequence from the MiXCR output table using the parameter [`-nFeature <gene_feature> germline`](https://mixcr.com/mixcr/reference/mixcr-export/#common-fields).
